prediction-insights.ipynb

```diff
@@ -0,0 +1,774 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Prediction Insights"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Overview\n",
+    "\n",
+    "Howso Engine enables powerful predictions with complete attribution and detailed explanations to make learning from\n",
+    "and debugging your data and predictions as easy as possible. For more information on predictions with Howso Engine,\n",
+    "check out the [predictions user guide](https://docs.howso.com/user_guide/basics/predictions.html)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import pandas as pd\n",
+    "from pmlb import fetch_data\n",
+    "\n",
+    "from howso.engine import Trainee\n",
+    "from howso.utilities import infer_feature_attributes"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Setup\n",
+    "\n",
+    "The [basic workflow guide](https://docs.howso.com/user_guide/basics/basic_workflow.html) goes into more specifics about the individual details of this section. This recipe will focus more on the insights.\n",
+    "\n",
+    "### Step 1: Load Data and Create Trainee"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "The following parameters from configuration file will override the Amalgam parameters set in the code: {'trace'}\n"
+     ]
+    }
+   ],
+   "source": [
+    "df = fetch_data(\"iris\", local_cache_dir=\"data\")\n",
+    "features = infer_feature_attributes(df)\n",
+    "\n",
+    "t = Trainee(features=features)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Step 2: Train, Analyze, and React"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "t.train(df)\n",
+    "t.analyze()\n",
+    "t.react_into_features(similarity_conviction=True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Step 3: React to a Case\n",
+    "\n",
+    "We supply a case using the `context_features` and `context_values` parameters. These are treated as the independent\n",
```

Comment:
You probably could have a higher level, more accessible definition, such as: The context features and values are the independent variables (inputs) from which you want to derive an output/prediction, which we call action features. Honestly, you may not need to add this at all. We could try to abstract all of the trainee building and predictions into a single step: Make a prediction, and then all the other steps are about gaining insights.